cmd/boost/deal_cmd.go

```diff
@@ -0,0 +1,344 @@
+	"github.com/filecoin-project/lotus/api"
+	chain_types "github.com/filecoin-project/lotus/chain/types"
+	lcli "github.com/filecoin-project/lotus/cli"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/google/uuid"
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/multiformats/go-multiaddr"
+	cli "github.com/urfave/cli/v2"
+
+	inet "github.com/libp2p/go-libp2p-core/network"
+)
+
+const DealProtocolv120 = "/fil/storage/mk/1.2.0"
+
+var dealCmd = &cli.Command{
+	Name:  "deal",
+	Usage: "Make an online deal with Boost",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "repo",
+			Usage: "repo directory for Boost client",
+			Value: "~/.boost-client",
+		},
+		&cli.StringFlag{
+			Name:     "http-url",
+			Usage:    "http url to CAR file",
+			Required: true,
+		},
+		&cli.StringSliceFlag{
+			Name:  "http-headers",
+			Usage: "http headers to be passed with the request",
+		},
+		&cli.StringFlag{
+			Name:     "provider",
+			Usage:    "provider address",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "commp",
+			Usage:    "commp of the CAR file",
+			Required: true,
+		},
+		&cli.Uint64Flag{
+			Name:     "piece-size",
+			Usage:    "size of the CAR file as a padded piece",
+			Required: true,
+		},
+		&cli.Uint64Flag{
+			Name:     "car-size",
+			Usage:    "size of the CAR file",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "payload-cid",
+			Usage:    "root CID of the CAR file",
+			Required: true,
+		},
+		&cli.IntFlag{
+			Name:  "start-epoch",
+			Usage: "",
+		},
+		&cli.IntFlag{
+			Name:  "duration",
+			Usage: "epochs for this deal to be active for",
+			Value: 518400, // default is 2880 * 180 == 180 days
+		},
+		&cli.IntFlag{
+			Name:  "provider-collateral",
+			Usage: "provider collateral required from storage miner",
+		},
+		&cli.Int64Flag{
+			Name:  "storage-price-per-epoch",
+			Usage: "",
+			Value: 1,
+		},
+		&cli.BoolFlag{
+			Name:  "verified",
+			Usage: "whether deal should be verified or not",
+			Value: true,
+		},
+	},
+	Before: before,
+	Action: func(cctx *cli.Context) error {
+		ctx := lcli.ReqContext(cctx)
+
+		sdir, err := homedir.Expand(cctx.String("repo"))
+		if err != nil {
+			return err
+		}
+
+		n, err := clinode.Setup(ctx, sdir)
+		if err != nil {
+			return err
+		}
+
+		api, closer, err := lcli.GetGatewayAPI(cctx)
+		if err != nil {
+			return fmt.Errorf("cant setup gateway connection: %w", err)
+		}
+		defer closer()
+
+		walletAddr, err := n.Wallet.GetDefault()
+		if err != nil {
+			return err
+		}
+
+		log.Debugw("selected wallet", "wallet", walletAddr)
+
+		maddr, err := address.NewFromString(cctx.String("provider"))
+		if err != nil {
+			return err
+		}
+		minfo, err := api.StateMinerInfo(ctx, maddr, chain_types.EmptyTSK)
+		if err != nil {
+			return err
+		}
+		if minfo.PeerId == nil {
+			return fmt.Errorf("could not find peer ID for stoarge provider %s", maddr)
```

Comment:
I actually prefer the way you phrased it in the error message below about no multiaddresses:
```suggestion
			return fmt.Errorf("storage provider %s has no peer ID set on-chain", maddr)
```